File: Dart/bubbleSort.dart
main() {
  bubbleSort([173, 657, 1, 0]);
}

void bubbleSort(List list) {
  if (list == null || list.length == 0) return;

  int n = list.length;
  int i, step;
  for (step = 0; step < n; step++) {
    for (i = 0; i < n - step - 1; i++) {
      if (list[i] > list[i + 1]) {
        swap(list, i);
      }
    }
  }
  print(list);
}

void swap(List list, int i) {
  int temp = list[i];
  list[i] = list[i + 1];
  list[i + 1] = temp;
}

File: javascript/bubble_sort.js
//Javascript function to implement Bubble Sort [Inplace Sorting Algorithm].
function bubble_Sort(a)
{
    var swapp;
    var n = a.length-1;
    var x=a;
    do {
        swapp = false;
        for (var i=0; i < n; i++)
        {
            if (x[i] < x[i+1])
            {
               var temp = x[i];
               x[i] = x[i+1];
               x[i+1] = temp;
               swapp = true;
            }
        }
        n--;
    } while (swapp);
 return x; 
}
//Sample example to test the function
console.log("Input array:");
console.log("[12, 345, 4, 546, 122, 84, 98, 64, 9]");
console.log("Sorted array using Bubble Sort:");
console.log(bubble_Sort([12, 345, 4, 546, 122, 84, 98, 64, 9]));
